tests/Unit/NumericalAlgorithms/Spectral/Test_ChebyshevGauss.cpp

```diff
@@ -0,0 +1,81 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+namespace Spectral {
+// To test the \f$w_k\f$ as opposed to the Spectral::quadrature_weights. We
+// don't provide a public accessor for the former since we only ever need them
+// in Spectral.cpp.
+template <Basis, Quadrature>
+std::pair<DataVector, DataVector> compute_collocation_points_and_weights(
+    size_t) noexcept;
+}  // namespace Spectral
+
+namespace {
+
+void test_points_and_weights(const size_t num_points,
+                             const DataVector expected_points,
+                             const DataVector expected_weights) {
+  const auto& points =
+      Spectral::collocation_points<Spectral::Basis::Chebyshev,
+                                   Spectral::Quadrature::Gauss>(num_points);
+  CHECK_ITERABLE_APPROX(expected_points, points);
+  // We test the \f$w_k\f$ here directly. Test_Spectral.cpp and
+  // Test_DefiniteIntegral.cpp take care of testing the
+  // Spectral::quadrature_weights.
+  const auto weights =
+      Spectral::compute_collocation_points_and_weights<
+          Spectral::Basis::Chebyshev, Spectral::Quadrature::Gauss>(num_points)
+          .second;
+  CHECK_ITERABLE_APPROX(expected_weights, weights);
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Numerical.Spectral.ChebyshevGauss.PointsAndWeights",
+                  "[NumericalAlgorithms][Spectral][Unit]") {
+  // Compare Chebyshev-Gauss points to Mathematica code available on request
```

Comment:
could your code to do this be written in python instead of Mathematica?

Reply:
Yes. In this case these are simple analytic expressions though, so I just added them to the comment with a reference.